src/plugin/plugin-client.js

```diff
@@ -87,11 +87,12 @@ class PluginClient extends EventEmitter {
     return this.deferredReply.promise;
   }
 
-  sendNotification(methodType, data) {
+  sendNotification(methodType, data, id) {
```

Comment:
I don't see a reason to add an additional field to every message.

Reply:
If Id is undefined, ```JSON.stringify``` which is called on sendJson clears id.
Is this tricky?
Is it better to change like following?
```js
const json = {
       messageType: methodType,
       data: data,
};
if(typeof id !== 'undefined'){
  json.id = id;
}
```

Reply:
Yes, I would prefer not to rely on that.

In another comment, I indicated that I didn't see a need for the ID anyway, so maybe this just goes away. If not, you should probably just add it to `data`.

Reply:
OK. I will remove id tomorrow.